sdk/openfeature-nodejs-provider/src/DevCycleProvider.ts

```diff
@@ -0,0 +1,277 @@
+        this.logger = options.logger ?? dvcDefaultLogger({ level: options.logLevel })
+    }
+
+    /**
+     * Generic function to retrieve a DVC variable and convert it to a ResolutionDetails.
+     * @param flagKey
+     * @param defaultValue
+     * @param context
+     * @private
+     */
+    private async getDVCVariable<I extends VariableValue, O>(
+        flagKey: string,
+        defaultValue: I,
+        context: EvaluationContext
+    ): Promise<ResolutionDetails<O>> {
+        const dvcVariable = this.dvcClient.variable(
+            this.dvcUserFromContext(context),
+            flagKey,
+            defaultValue
+        )
+        return this.resultFromDVCVariable<O>(
+            dvcVariable instanceof Promise
+                ? await dvcVariable
+                : dvcVariable
+        )
+    }
+
+    /**
+     * Resolve a boolean OpenFeature flag and its evaluation details.
+     * @param flagKey
+     * @param defaultValue
+     * @param context
+     */
+    async resolveBooleanEvaluation(
+        flagKey: string,
+        defaultValue: boolean,
+        context: EvaluationContext
+    ): Promise<ResolutionDetails<boolean>> {
+        return this.getDVCVariable(flagKey, defaultValue, context)
+    }
+
+    /**
+     * Resolve a string OpenFeature flag and its evaluation details.
+     * @param flagKey
+     * @param defaultValue
+     * @param context
+     */
+    async resolveStringEvaluation(
+        flagKey: string,
+        defaultValue: string,
+        context: EvaluationContext
+    ): Promise<ResolutionDetails<string>> {
+        return this.getDVCVariable(flagKey, defaultValue, context)
+    }
+
+    /**
+     * Resolve a number OpenFeature flag and its evaluation details.
+     * @param flagKey
+     * @param defaultValue
+     * @param context
+     */
+    async resolveNumberEvaluation(
+        flagKey: string,
+        defaultValue: number,
+        context: EvaluationContext
+    ): Promise<ResolutionDetails<number>> {
+        return this.getDVCVariable(flagKey, defaultValue, context)
+    }
+
+    /**
+     * Resolve a object OpenFeature flag and its evaluation details.
+     * @param flagKey
+     * @param defaultValue
+     * @param context
+     */
+    async resolveObjectEvaluation<T extends JsonValue>(
+        flagKey: string,
+        defaultValue: T,
+        context: EvaluationContext
+    ): Promise<ResolutionDetails<T>> {
+        return this.getDVCVariable(flagKey, this.defaultValueFromJsonValue(defaultValue), context)
+    }
+
+    /**
+     * Convert a OpenFeature JsonValue default value into DVCJSON default value for evaluation.
+     * @param jsonValue
+     * @private
+     */
+    private defaultValueFromJsonValue(jsonValue: JsonValue): DVCJSON {
+        if (typeof jsonValue !== 'object' || Array.isArray(jsonValue)) {
+            throw new Error('DevCycle only supports object values for JSON flags')
```

Comment:
If you'd like, you could throw a [InvalidContextError](https://github.com/open-feature/js-sdk/blob/main/packages/shared/src/errors/invalid-context-error.ts) here, which will cause the SDK to set the error code accordingly and potentially help with debugging.